### PR TITLE
Issue #1405 Invalid index reading 'raw' message

### PR DESCRIPTION
**Pull request by trink**

### CODE REVIEW

File: pipeline/pipeline_runner.go [hunk 1]
```diff
@@ -158,7 +158,7 @@ type PipelinePack struct {
 // Returns a new PipelinePack pointer that will recycle itself onto the
 // provided channel when a message has completed processing.
 func NewPipelinePack(recycleChan chan *PipelinePack) (pack *PipelinePack) {
-	msgBytes := make([]byte, message.MAX_MESSAGE_SIZE)
+	msgBytes := make([]byte, 0, message.MAX_MESSAGE_SIZE)
```
Comment (rafrombrc) [inline pipeline/pipeline_runner.go]: This worries me a bit. `pack.Zero()` sets MsgBytes length to the full capacity, and our code generally copies into MsgBytes and then resizes down, rather than resizing up and then copying. I'm okay w/ changing this, but I'd rather do it in a separate PR, since it requires digging into all of the places where we set pack.MsgBytes to make sure we don't silently truncate b/c the length is too short.

Comment (trink) [inline pipeline/pipeline_runner.go]: "our code generally copies into MsgBytes and then resizes down" Can you point me to where you are referring, The length is meaningless when populating it with something new and we should only be dealing with capacity (like this):

```Go
		messageLen := len(unframed)
		if messageLen > cap(pack.MsgBytes) {
			pack.MsgBytes = make([]byte, messageLen)
		}
		pack.MsgBytes = pack.MsgBytes[:messageLen]
		copy(pack.MsgBytes, unframed)
```
Also if read_message("raw") is called on a new empty message it should not return MAX_MESSAGE_SIZE of NUL bytes (which is the reason the change is included here)

Comment (rafrombrc) [inline pipeline/pipeline_runner.go]: One [place](https://github.com/mozilla-services/heka/blob/dev/pipeline/pipeline_runner.go#L215) where it happens is my fault, it just landed. I'm pretty sure I tried to make this change earlier and hit a side-effect somewhere else, too, but I just searched through the code and couldn't find anything.

I'm fine w/ trying this out, but the code I linked to above should be changed so the resize happens before the copy, and the Zero method should be changed to match the new pack behavior.

**Comment by trink** [comment]
Have to make half that last change in 0.9 and the other in dev.  Closing and deleting this branch
